client/app/bundles/course/survey/containers/QuestionFormDialogue/QuestionFormOptions.jsx

```diff
@@ -34,69 +33,116 @@ const optionsTranslations = defineMessages({
   },
 });
 
-class QuestionFormOptions extends Component {
-  handleSelectFiles = (event) => {
-    const { fields } = this.props;
-    const options = fields.getAll();
-    const files = event.target.files;
+const handleSelectFiles = (event, fieldsConfig) => {
+  const { append } = fieldsConfig;
+  const files = event.target.files;
 
-    // eliminate blank options
-    fields.removeAll();
-    options.forEach((option) => {
-      if (option.option || option.file) {
-        fields.push(option);
-      }
+  for (let i = 0; i < files.length; i += 1) {
```

Comment:
Can do this in a functional style:

```js
files.forEach((file) => append({ ... }));
```

Reply:
files here is of FileList type, so we can do it in the functional style.